tools/monorepo-utils/src/changelog/lib/github.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * Internal dependencies
+ */
+import { getPullRequest, isCommunityPullRequest } from '../../core/github/repo';
+import { Logger } from '../../core/logger';
+
+/**
+ * Get relevant data from a pull request.
+ *
+ * @param {Object} options
+ * @param {string} options.owner repository owner.
+ * @param {string} options.name  repository name.
+ * @param {string} prNumber      pull request number.
+ * @return {Promise<object>}     pull request data.
+ */
+export const getPullRequestData = async (
+	options: { owner: string; name: string },
+	prNumber: string
+) => {
+	const { owner, name } = options;
+	const prData = await getPullRequest( { owner, name }, prNumber );
+	const isCommunityPR = isCommunityPullRequest( prData, owner, name );
+	const headOwner = isCommunityPR ? prData.head.repo.owner.login : owner;
+	const branch = prData.head.ref;
+	const fileName = branch.replace( /\//g, '-' );
+	const prBody = prData.body;
+	const head = prData.head.sha;
+	const base = prData.base.sha;
+	return {
+		prBody,
+		isCommunityPR,
+		headOwner,
+		branch,
+		fileName,
+		head,
+		base,
+	};
+};
+
+/**
+ * Determine if a pull request description activates the changelog automation.
+ *
+ * @param {string} body pull request description.
+ * @return {boolean} if the pull request description activates the changelog automation.
+ */
+export const getShouldAutomateChangelog = ( body: string ) => {
```

Comment:
Not sure whether / where we have naming conventions documented, but for boolean getters I've usually seen (in other projects) the `get` omitted to then have it start with something like the `should` here, or `will` or `does` or `is` or something similar.

Reply:
yeah what a conundrum! I went back and forth on this one a few times and settled on this so that I could have the const appropriately named in the code:

```js
const shouldAutomateChangelog = getShouldAutomateChangelog( body );
``` 
If the function is called `shouldAutomateChangelog`, what do I call the const? Maybe `willAutomateChangelog` for the function?

Reply:
Hmmm, when taking a principled vantage point I'd say that it shouldn't be the responsibility of a module how its callers will cache its results. 

Pragmatically speaking, I'd probably not even assign it to a const at all in index.ts, as it's used only once. 

Another pragmatic stance could be to name the const `_shouldAutomateChangelog`, even though that has implications of class membership in other contexts. 

All in all I'd say you've taken a pragmatic approach as well that is completely fine, and I'm happy to have things move forward with this or one of the other mentioned solutions. ☺️